api/types/types.go

```diff
@@ -319,19 +319,22 @@ type ContainerJSONBase struct {
 	SizeRootFs      *int64 `json:",omitempty"`
 }
 
-// ContainerJSON is newly used struct along with MountPoint
+// ContainerJSON represents the state and settings of a container when inspecting it.
 type ContainerJSON struct {
 	*ContainerJSONBase
 	Mounts          []MountPoint
 	Config          *container.Config
 	NetworkSettings *NetworkSettings
 }
 
-// NetworkSettings exposes the network settings in the api
+// NetworkSettings holds the networking state of a specific container when inspecting it.
 type NetworkSettings struct {
-	NetworkSettingsBase
-	DefaultNetworkSettings
-	Networks map[string]*network.EndpointSettings
+	NetworkSettingsBase                // Deprecated: DefaultNetworkSettings is deprecated since API v1.21 and will be removed in a future release.
+	DefaultNetworkSettings             // Deprecated: DefaultNetworkSettings is deprecated since API v1.21 and will be removed in a future release.
+	SandboxID              string      // SandboxID uniquely represents a container's network stack
+	SandboxKey             string      // SandboxKey is the full path of the netns handle
+	Ports                  nat.PortMap // Ports is a collection of PortBinding indexed by Port
```

Comment:
Duplicating these fields is a breaking change; the insidious kind of breaking change that doesn't break compilation.
```go
myNetworkSettings.SandboxID != myNetworkSettings.NetworkSettingsBase.SandboxID
```
I'm afraid that `NetworkSettingsBase` can't be deprecated or removed until we migrate to Go modules and release a new major version (read: with a `/vN` module path). We can still deprecate the fields inside `NetworkSettingsBase` at least, and drop the `NetworkSettingsBase` type without any ceremony in a new module version.

Reply:
For now, we'll continue to write both fields so I don't think that's gonna be a problem. In a release or two, we should think about removing the whole `NetworkSettingsBase` struct, or at least the fields that were copied to `NetworkSettings` (ie. `Bridge`, `SandboxID`, `SandboxKey`). This should give enough time for our Go users to migrate to the new fields.

Reply:
[Yes, it's gonna be a problem.](https://go.dev/play/p/p4cUgBV5TpE)

Reply:
Mh right. So, is it okay if I just remove the fields I added to `NetworkSettings` and keep the deprecation warning on NetworkSettingsBase while mentioning that non-deprecated fields will be moved to `NetworkSettings` in v26?

Reply:
That'd work, so long as linters don't complain about references to fields like `myNetworkSettings.SandboxID` which would be well-formed in both v25 and v26. The deprecation warning on the struct itself might have to go.